src/pages/SelectedVideoPage/VideoPage.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useRef, useEffect } from "react";
+import { useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
+import ReactPlayer from "react-player/youtube";
+import styles from "./VideoPage.module.css";
+import { VideoSuggestionCard } from "components";
+import { videoShufflerFunction } from "utilities";
+
+const VideoPage = () => {
+  const { videoId } = useParams();
+  const { videosList } = useSelector((storeReceived) => storeReceived.videos);
+  const { sidebarView } = useSelector((storeReceived) => storeReceived.sidebar);
+  const videoSelected = videosList.find(
+    (everyVideo) => everyVideo._id === videoId
+  );
+  const { title, dateOfUpload, viewQuantity, description } = videoSelected;
+  const videoPageRef = useRef();
+  const [videoPageDimensions, setVideoPageDimensions] = useState();
+  const getCurrentPageDimensions = () => {
+    const widthReceived = videoPageRef.current.clientWidth;
+    setVideoPageDimensions(widthReceived);
+  };
+  //   For receiving the page width whenever the window is resized and setting it
+  useEffect(() => {
+    getCurrentPageDimensions();
+  }, [videosList, sidebarView]);
+  useEffect(() => {
+    window.addEventListener("resize", getCurrentPageDimensions);
+  }, []);
+  return (
```

Comment:
can't we use one useeffect instead of two?

Reply:
Yes we can but I have splitted it into 2 parts to avoid confusion that might occur if I call the function as well and at the same time add it to the window as an event listener too.